### PR TITLE
Feature/cancellation

### PR DESCRIPTION
**Pull request by corinnaj**
Add the option to provide a CancellationToken to immediately stop SpeechOut.

### CODE REVIEW

File: Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs [hunk 1]
```diff
@@ -2,32 +2,38 @@
...
+        Dialog(token);
     }
 
-    public async void Dialog(){
-        await speechOut.Speak("This is a sample speech output script.");
-        await speechOut.Speak("If you hear my voice, the speech out system is functional.");
-        await speechOut.Speak("You may hear me in a different accent. If that's the case, please check your OS speech settings.");
+    public async void Dialog(CancellationToken token)
+    {
+        await speechOut.Speak("This is a sample speech output script.", token: token);
```
Comment (JotaroS) [inline Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs]: here I __really__ want a back compatibility: maybe keep the Speak() without `token` ?

**Review by JotaroS — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs (hunk 1 above)
Comment (corinnaj) [reply Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs]: The token is an optional parameter, you don't need to provide it. (For example the Dialog Scene, does not provide a token for speech)

Comment (JotaroS) [reply Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs]: I see, then I won't put that in this example scripts: I intended to provide the most simple use of `speechOut.Speak()`. Maybe can you make a new example scene/script that allows cancelation?

**Review by JotaroS — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs (hunk 1 above)
Comment (corinnaj) [reply Assets/Scripts/Example Speech Scripts/MySpeechOutScript.cs]: Alright, I created a new sample scene and script for cancelling voice output

**Review by JuliusCosmoRomeo — APPROVED** [review]

**Comment by JotaroS** [comment]
@JuliusCosmoRomeo hey, did you test if the code works for others? (maybe ask Deb for that. We really need to check if the framework works both on windows / mac). Thanks.

**Pull request merged**